lib/iris/util.py

```diff
@@ -2059,31 +2059,53 @@ def equalise_attributes(cubes):
     See more at :doc:`/userguide/real_and_lazy_data`.
 
     """
+    # deferred imports to avoid circularity problem
+    from iris.common._split_attribute_dicts import (
+        _convert_splitattrs_to_pairedkeys_dict,
+    )
+    from iris.cube import CubeAttrsDict
+
+    cube_attrs = [cube.attributes for cube in cubes]
     removed = []
+
+    # *IF* they are cube attributes, then convert to "paired-keys" form.
+    is_split_dicts = any(
+        isinstance(attrs, CubeAttrsDict) for attrs in cube_attrs
+    )
+    if is_split_dicts:
```

Comment:
This is already handled by your addition to `_convert_splitattrs_to_pairedkeys_dict()`:

https://github.com/SciTools/iris/blob/35ec095a6293394c599c71a8238db5f10e2f9e13/lib/iris/common/_split_attribute_dicts.py#L39-L41

So can we get away without any checks at either end?

Reply:
The original point here was the the conversion to/from paired-keys type is only needed for split-type attributes
(which technically we should _always_ have, if it is only used on cubes, but I still think it's safer to not insist on that)

On reflection it should still _work_, if less efficient, to do this with regular attributes dictionaries too -- i.e. always + regardless of type.
I will try that, as it clearly would make the code simpler.

Reply:
Yup, it works !
https://github.com/SciTools/iris/pull/5586/commits/c3e2175b7ea7735ed8ab2b49d926d70922a4afde